### PR TITLE
Create networking-console-plugin.md

### PR DESCRIPTION
**Pull request by upalatucci**

### CODE REVIEW

File: enhancements/network/networking-console-plugin.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+N/A
+
+### Drawbacks
+
+This means extra work in the beginning to integrate the operatand with the operator. 
+
+## Test Plan
+
+E2e tests will run at every pr on the networking-console-plugin using prow just like other console dynamic plugins
```
Comment (spadgett) [inline enhancements/network/networking-console-plugin.md:106]: Do we need other periodic tests to run to make sure changes in console don't break the plugin?

File: enhancements/network/networking-console-plugin.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+N/A
+
+## Proposal
+
+To execute the migration of networking console pages to the "networking-console-plugin" repository and integrate them into the cluster-networking-operator, we propose the following steps:
+
+Create "networking-console-plugin" Repository (Done): Establish a new repository named "networking-console-plugin" to house the codebase for networking console pages. This repository will serve as the centralized location for all networking-related functionalities.
+
+Move Networking Console Pages (Done): Transfer relevant code files and resources from the console repository to the "networking-console-plugin" repository and refactor them. Ensure accurate migration of dependencies and configurations to maintain seamless functionality.
```
Comment (spadgett) [inline enhancements/network/networking-console-plugin.md:64]: How does this change impact the topology view where we have a details sidebar for networking resources like services and routes?

@jerolimov FYI

File: enhancements/network/networking-console-plugin.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+
+
+## Summary
+
+
+At present, several pages within the networking section of our console are defined within the console repository. 
+However, working with the console repository has slowed down development and there are multiple motivations to convert static console plugins into dynamic ones described [here](https://github.com/spadgett/enhancements/blob/master/enhancements/console/dynamic-plugins.md#motivation). 
+To address this, we propose relocating the relevant code to a separate repository named "networking-console-plugin" and integrating it into the cluster-networking-operator.
+Specifically: Services, Route, Ingress, NetworkPolicy, and NetworkAttachmentDefinition pages for listing, creating, and editing those resources.
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply enhancements/network/networking-console-plugin.md:33]: Yes we currently implementing those features and merged [one PR](https://github.com/openshift/console/pull/13722) in the console to close those gaps. The pr add some useful functionality even for other plugins, not just this one.

File: enhancements/network/networking-console-plugin.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+Create "networking-console-plugin" Repository (Done): Establish a new repository named "networking-console-plugin" to house the codebase for networking console pages. This repository will serve as the centralized location for all networking-related functionalities.
+
+Move Networking Console Pages (Done): Transfer relevant code files and resources from the console repository to the "networking-console-plugin" repository and refactor them. Ensure accurate migration of dependencies and configurations to maintain seamless functionality.
+
+Integration with Cluster-Networking-Operator: Modify the cluster-networking-operator to incorporate the networking console plugin as an operand. Update configurations and references accordingly to seamlessly integrate the networking pages with the operator.
+
+Incorporation into OpenShift Release Payload: Upon successful migration and integration, include the networking console plugin in the OpenShift release payload. This ensures the plugin is delivered by default with the console, providing users immediate access to enhanced networking functionalities.
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply enhancements/network/networking-console-plugin.md:68]: The operator that manage the plugin resources, will enable that. 
Monitoring operator is doing the same for the monitoring plugin [here](https://github.com/openshift/cluster-monitoring-operator/blob/master/pkg/client/client.go#L1710). 
Untile we have "Red Hat signed" plugins there is no other solution at the moment.

File: enhancements/network/networking-console-plugin.md (hunk 1 above)
Comment (upalatucci) [reply enhancements/network/networking-console-plugin.md:106]: I think UI-related changes are the most important things to test in those cases. Sometimes even a small change in the console CSS can change things in plugins drastically. But unfortunately, those kinds of tests are difficult to implement. 
But yeah in general i think it's a good idea even for other plugins

File: enhancements/network/networking-console-plugin.md (hunk 2 above)
Comment (upalatucci) [reply enhancements/network/networking-console-plugin.md:64]: At the moment we don't impact this part. We just re-define the list, details and creation pages

Comment (upalatucci) [reply enhancements/network/networking-console-plugin.md:64]: we have a [pr](https://github.com/openshift/console/pull/13675) in place to remove those pages from the console code.

File: enhancements/network/networking-console-plugin.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (spadgett) [reply enhancements/network/networking-console-plugin.md:68]: > The operator that manage the plugin resources, will enable that.

That's reasonable. We should mention it in the enhancement.

File: enhancements/network/networking-console-plugin.md (hunk 1 above)
Comment (spadgett) [reply enhancements/network/networking-console-plugin.md:106]: It would be good to cover this somehow in the test plan section.

**Pull request closed without merging**